core/app/models/workarea/checkout.rb

```diff
@@ -161,7 +161,9 @@ def continue_as(user)
     # @return [self]
     #
     def update(params = {})
-      steps.each { |s| s.new(self).update(params) }
+      steps.reduce(true) do |_, step|
+        step.new(self).update(params)
```

Comment:
This will only ever return the result of the last step.

Reply:
Yeah I'm sorta confused about the whole way this works. When I print out all the update results, I get `false` for almost every one and then `true` for payment. But if I add `result && step.new(self).update(params)`, my assertion for whether checkout updates does not pass, because `result` is going to be false, so it never actually runs update for any of the steps.